mitodl/open-discussions
# Improved performance of /learn APIs

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #2692
Fixes #2691

#### What's this PR do?
This PR does two things primarily:
- Adds some indexes that improve the performance of queries on the postgres side, based on running `ANALYZE` on the queries django is generating.
- Altered the queries to avoid some extra joins and N+1 queries in favor of annotations and prefetching.

#### How should this be manually tested?
Go to /learn and ensure the page loads, is reasonably performant, and the data matches what you get back on the `master` branch.
